power_grid_model_c/power_grid_model/include/power_grid_model/math_solver/y_bus.hpp

```diff
@@ -350,6 +354,79 @@ template <bool sym> class YBus {
         admittance_ = std::make_shared<ComplexTensorVector<sym> const>(std::move(admittance));
     }
 
+    template <UpdateIsProgressive T>
+    void update_admittance_increment(std::shared_ptr<T const> const& math_model_param_incrmt,
```

Comment:
i think you can split this entire algorithm into a couple separate functions. they probably even have a lot in common so you might be able to remove duplication

Reply:
function refactored as suggested